### PR TITLE
Add cluster ID to clock

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Add cluster ID to clock object

**Why?**
Business logic can tell the difference (clock generated by cluster a vs cluster b)

**How did you test it?**
Existing tests

**Potential risks**
N/A

**Is hotfix candidate?**
N/A

### CODE REVIEW

File: common/cluster/metadata.go [hunk 1]
```diff
@@ -59,6 +59,8 @@ type (
 		IsGlobalNamespaceEnabled() bool
 		// IsMasterCluster whether current cluster is master cluster
 		IsMasterCluster() bool
+		// GetClusterID return the cluster ID, which is also the initial failover version
+		GetClusterID() int64
```
Comment (wxing1292) [inline common/cluster/metadata.go:63]: @yiminc @yycptt @yux0 another solution is to use the UUID from cluster metadata table, what do you think?

Comment (yiminc) [reply common/cluster/metadata.go:63]: I think UUID is overkill. Initial failover should be good.

File: proto/internal/temporal/server/api/clock/v1/message.proto [hunk 2]
```diff
@@ -24,7 +24,8 @@ package temporal.server.api.clock.v1;
 
 option go_package = "go.temporal.io/server/api/clock/v1;clock";
 
-message ShardClock {
-    int32 id = 1;
+message Clock {
```
Comment (yiminc) [inline proto/internal/temporal/server/api/clock/v1/message.proto:27]: I think we should keep ShardClock. Or use VectorClock.

**Review by yiminc — APPROVED** [review]

**Review by yycptt — APPROVED** [review]

File: service/history/vclock/vclock.go [hunk 3]
```diff
@@ -54,14 +56,16 @@ func HappensBefore(
 }
 
 func Compare(
-	clock1 *clockspb.ShardClock,
-	clock2 *clockspb.ShardClock,
+	clock1 *clockspb.Clock,
+	clock2 *clockspb.Clock,
 ) (int, error) {
-	if clock1.GetId() != clock2.GetId() {
+	if clock1.GetClusterId() != clock2.GetClusterId() || clock1.GetShardId() != clock2.GetShardId() {
```
Comment (yycptt) [inline service/history/vclock/vclock.go:62]: We probably need to differentiate those two cases? ShardID not equal is unexpected, while clusterID not equal is expected in some cases (due to auto-forwarding).

Comment (wxing1292) [reply service/history/vclock/vclock.go:62]: good point, actually even without the cluster ID, clocks generated by different cluster cannot be compared

**Pull request merged**